### DISCUSSION TITLE
How to handle initialization of more than 16 API groups in OpenAPI module

### DISCUSSION THREAD

**Discussion by gudaoxuri** [poem-web/poem#273] [Q&A] [answered]
When using the following method, an error will be reported when the number of API groups is greater than 16.

```rust
let api_service =OpenApiService::new((Api1,Api2,...), "Todos", "1.0.0").server("http://localhost:3000");
let ui = api_service.swagger_ui();
let spec = api_service.spec();
```

https://github.com/poem-web/poem/blob/008679da69f3943c52b8ce858b9a5957140c74c9/poem-openapi/src/base.rs#L392-L400

In the actual system, it is common to have more than 16 API groups. Please tell me how to handle this requirement in poem, thanks!

**Comment by sunli829** [comment, marked as answer]
This tuple can be nested.

```rust
((Api1, Api2,... Api16), (Api17, Api18,... Api32), ...)
```

**Reply by gudaoxuri** [reply]
Nice!
